### PR TITLE
🏗📖 Auto-format all markdown files in `ampproject/amphtml` with Prettier

### LINKED ISSUE #25148: Shall we auto-format our `.md` files with Prettier?
Today, we use [Prettier](https://prettier.io/) to [format and fix](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#code-quality-and-style) all our JS files and several non-JS files. The format checker has [supported](https://prettier.io/blog/2017/11/07/1.8.0.html) Markdown files for a while now.

**Question:** Shall we start auto-formatting our `.md` files with Prettier?

**Pros:**
- One [single standard](https://prettier.io/docs/en/index.html) for the internal structure of documentation files
- IDE support for [auto-fix-on-save](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#workflow-for-visual-studio-code) while editing code
- Command line support for [auto-fixing local changes](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#manually-fixing-code)
- Possible to [override](https://prettier.io/docs/en/ignore.html#markdown) Prettier's auto-format behavior for sections of code
- Possible to [auto-wrap](https://prettier.io/docs/en/options.html#prose-wrap) text for easy editing

**Cons:**
- There might be some small changes to how `.md` files look when they are rendered
    - Attribute lists are changed to [all-in-one-line](https://github.com/ampproject/amphtml/compare/master...rsimha:2019-10-18-PrettifyMarkdown#diff-8e253a89bfca066fd8ca624c28468261R295) if they all fit on one line
image
    - Attribute lists are changed to [one-per-line](https://github.com/ampproject/amphtml/compare/master...rsimha:2019-10-18-PrettifyMarkdown#diff-5b53438fd782012c7289a8bb5107d86b) if they don't all fit on one line
image

- Failing `gulp prettify` checks can be annoying if an occasional contributor makes a badly formatted change
- No easy way to auto-format code with Prettier if you edit code with the GitHub web-UI

@CrystalOnScript @mrjoro @pbakaus @ampproject/wg-outreach Thoughts?

### PR DESCRIPTION
**Pull request by rsimha**
The AMP Project already uses [Prettier](https://prettier.io/) to [auto-format](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#code-quality-and-style) JS files and several non-JS files across the repo. This PR adds documentation files to the mix. 

**Highlights:**
- Includes `.md` files in the list of files that are checked by Prettier
- Auto-formats all `.md` files in the repo by running `gulp prettier --fix`
- Skips the broken links check for PRs that modify a large number of `.md` files
- Prints suggested fixes when formatting errors are detected
- Exempts GitHub issue templates (they use a custom header that's incompatible with Prettier)
- Make extension generator (`gulp make-extension`) resilient to markdown file auto-fixing

**Notes:**
- Most of these changes do not affect the rendered HTML. Exceptions are:
    - Attribute lists are changed to all-in-one-line if they all fit on one line
image
    - Attribute lists are changed to one-per-line if they don't all fit on one line
image
- Once this PR is merged, `.md` files will have to satisfy Prettier's formatting requirements
- A workflow to make your IDE auto-fix `.md` files on save is available [here](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#workflow-for-visual-studio-code)
- An alternative is to manually run `gulp prettify --local_changes --fix`

**Example of a failing check:**

Fixes #25148

### CODE REVIEW

**Comment by rsimha** [comment]
So many [broken links](https://travis-ci.org/ampproject/amphtml/jobs/601024915#L1476) in our documentation :frowning_face:

**Comment by rsimha** [comment]
I sanity checked several of the changes in this PR, and found no visible changes to the rendered HTML. This is now ready for review.

**Comment by rsimha** [comment]
Rebased and added a function to print suggested fixes for badly formatted `.md` files.

File: .github/ISSUE_TEMPLATE/bug-report.md [hunk 1]
```diff
@@ -4,7 +4,6 @@ about: Used to report bugs in AMP.
 title: ''
 labels: 'Type: Bug'
 assignees: ''
-
```
Comment (mrjoro) [inline .github/ISSUE_TEMPLATE/bug-report.md]: are we sure the issue templates don't need this?

File: .github/ISSUE_TEMPLATE/cherry_pick_template.md [hunk 2]
```diff
@@ -10,26 +10,29 @@ MUST: Replace *everything* in angle brackets in the title AND body of this issue
...
 # Cherry-pick request
 
 <!--
 TIP: Cherry-picks into production most likely require a cherry-pick into RC too. Otherwise, your fix will be lost when the RC is promoted.
 -->
 
-| Issue | PR  | Production? | RC? | [Release issue](https://github.com/ampproject/amphtml/labels/Type%3A%20Release) |
-| :---: | :-: | :---------: | :-: | :-----------: |
-| #<_ISSUE_NUMBER_> | #<_PR_NUMBER_> | **<YES/NO>** | **<YES/NO>** | #<_RELEASE_ISSUE_> |
+|       Issue       |       PR       | Production?  |     RC?      | [Release issue](https://github.com/ampproject/amphtml/labels/Type%3A%20Release) |
```
Comment (mrjoro) [inline .github/ISSUE_TEMPLATE/cherry_pick_template.md]: What's going on with the spacing here?  I assumed it was trying to give the same space to each column, but the hyphens below don't look the same length.

File: 3p/README.md [hunk 3]
```diff
@@ -47,7 +47,7 @@ The following aren't hard requirements, but are performance optimizations we sho
 - Creatives should only animate using CSS transforms and opacity.
 - When creatives animate they should not use more than 2ms time per frame to allow for other animations to have sufficient time for concurrent animations.
 
-The better an ad network does on the above requirements, the earlier ads from it will be loaded into AMP. In other words: *The slower the ad loads and the more it interferes with the page, the later AMP will load it.*
+The better an ad network does on the above requirements, the earlier ads from it will be loaded into AMP. In other words: _The slower the ad loads and the more it interferes with the page, the later AMP will load it._
```
[the remark this answers is not in the document]
Comment (mrjoro) [reply 3p/README.md]: nm; I see both are italics and both are valid, so standardizing on one makes sense

**Comment by rsimha** [comment]
Rebased, reverted changes to issue template headers, and exempted them from future checks.

**Comment by rsimha** [comment]
Some final notes:
- Self-reviewed every file in `contributing/`, and most files in `ads/`, `extensions/`, etc.
- Found and fixed a handful of existing bugs that were exposed by auto-formatting (tabs instead of spaces, bad formatting, etc.)
- Added `prettier-ignore` overrides where required
- Verified that the majority of `.md` files have no visible changes when rendered
- Tested VSCode auto-fix-on-save workflow for `.md` files
- Tested `gulp prettify` with `--fix`, `--files`, and `--local_changes`

Merging this now :crossed_fingers:

**Pull request merged**